plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -250,16 +258,28 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
             return symbol.dispatchReceiverType ?: symbol.resolvedReceiverTypeRef?.type
         }
 
+    private fun FirPropertySymbol.embedMemberPropertyName() = this.callableId.embedMemberPropertyName(
+        Visibilities.isPrivate(this.visibility)
+    )
+
     /**
      * Construct and register the field embedding for this property's backing field, if any exists.
      */
-    private fun processBackingFields(symbol: FirPropertySymbol, embedding: ClassTypeEmbedding): Pair<SimpleKotlinName, FieldEmbedding>? {
+    @OptIn(SymbolInternals::class)
+    private fun processBackingFields(
+        symbol: FirPropertySymbol,
+        classSymbol: FirRegularClassSymbol
+    ): Pair<SimpleKotlinName, FieldEmbedding>? {
+        val embedding = embedClass(classSymbol)
         val unscopedName = symbol.callableId.embedUnscopedPropertyName()
-        val name = symbol.callableId.embedMemberPropertyName()
-        embedding.findAncestorField(unscopedName)?.let { return null }
-        val backingField = name.specialEmbedding() ?: symbol.hasBackingField.ifTrue {
+        val scopedName = symbol.embedMemberPropertyName()
+        val fieldIsAllowed = symbol.hasBackingField
+                && (symbol.getterSymbol?.fir is FirDefaultPropertyGetter ||
+                symbol.setterSymbol?.fir is FirDefaultPropertySetter)
```

Comment:
I think the logic here needs to be more complicated.  Let's say the getter is default but the setter is custom, then we should not use a field.  But if the getter is default and the setter is missing, we should.  Or is this covered somehow?

Reply:
I guess it is not that important how to deal with the cases when the getter is default and the setter is custom or vice versa. But I agree it can create unwanted behavior with current implementation (e.g. inhale access and then call the getter method).